test/unit/eigensolver/test_eigensolver.cpp

```diff
@@ -55,16 +64,23 @@ void testEigensolver(const blas::Uplo uplo, const SizeType m, const SizeType mb)
     return reference;
   }();
 
-  Matrix<T, D> mat_a(reference.distribution());
-  copy(reference, mat_a);
+  Matrix<T, Device::CPU> mat_a_h(reference.distribution());
+  copy(reference, mat_a_h);
 
-  auto ret = eigensolver::eigensolver<B>(uplo, mat_a);
+  auto ret = [&]() {
+    MatrixMirror<T, D, Device::CPU> mat_a(mat_a_h);
+    return eigensolver::eigensolver<B>(uplo, mat_a.get());
+    // TODO there is a superflous copy back
+  }();
```

Comment:
@msimberg follow up https://github.com/eth-cscs/DLA-Future/pull/513#discussion_r840617468

Reply:
IMO is not superfluous in this case. As the matrix is changed you want to copy back the new values, such that the effect in both cases.
There is no test here, but it is better to keep it consistent.